File: esponenziale calcolato/esponenziale_calcolato.java

import java.util.Scanner;


public class Main {

	public static void main(String[] args) {
		Main main=new Main();
		main.runtime();
		
	}
	public void runtime() {
		while(true) {
			double espreal=Math.exp(2);
			System.out.println("esponenziale reale,solo errore di arrotondamento:"+espreal);
			System.out.println("inserire parametro per l'esponente calcolato:");
			Scanner scanner=new Scanner(System.in);
			int k=scanner.nextInt();
			double esptot=0;
			for(int i=0;i<k;i++) {
				esptot=esptot+this.espcalc(i);
			}
		double error=Math.abs(espreal-esptot);
			System.out.println("l'esponenziale di due approssimato al k-esimo termine è:"+esptot+"\n l'errore analitico è:"+ error);
			
			
		}

		
	}
	public double espcalc(int index) {
		double espcal=Math.pow(2, index)/this.factorial(index);
		return espcal;
	}
	public double factorial(int index) {
		long fact=1;
		for(int j=2;j<=index;j++) {
			fact=fact*j;
		}
		return fact;
	}

}

File: esponenziale calcolato/esponenziale_calcolato.js
//calcolo esponenziale
var index=0
var espreal=Math.exp(2)
var exppar=document.getElementById("exppar")
var inputcamp=document.getElementById("input")
var result=document.getElementById("result")
exppar.innerText="esponenziale in algebra reale:"+espreal

function calcola(){
    index=parseInt(inputcamp.value)
    let appr_attuale=0
    for(let i=0;i<=index;i++){
        let indice_sommatoria=Math.pow(2,i)/factorial(i)
        appr_attuale=appr_attuale+indice_sommatoria
      
        
    }
    result.innerText="esponenziale calcolato al termine k-esimo della sommatoria="+appr_attuale.toFixed(30)+"   errore algoritmico="+(espreal-appr_attuale).toFixed(30)


}

function factorial(index){
    if(index<2){
        return 1
    }
    let j=1
    let factorial=1
    for(j=2;j<=index;j++){
        factorial=factorial*j

    }
    return factorial

}

